### PR TITLE
Datagen proposal to amend payment for Milestone 2

### PR DESCRIPTION
**Pull request by Lord-Nymphis**
Project Abstract

B-Datagray’s Datagen project concerns the development of a decentralized infrastructure for CPU/GPU cloud computing, in chain, through different blockchain components.

The Datagen infrastructure requires the creation of a Substrate-based blockchain with some key features: low latency time, high efficiency (compatibly with the decentralized nature of the network), high degree of decentralization (higher than a traditional PoS would allow) of the physical hardware providing the cloud computing (therefore requiring a customized consensus protocol) and in-chain (or near-in-chain) computation (no off-chain based solutions).

The Datagen infrastructure primarily want to serve: privacy friendly search engines and browsers, decentralized Web3 games and other ones (for example decentralizing the hardware layer of PoS blockchains, typically hosted on AWS and similar, etc…).

We will implement only a PoC with this grant.

The goal is to achive a fully functional mechanism for the random selection of the nodes in the fast blockchian and smooth communication between the two blockchains.

This pull is to propose a change in Milestone 2 amount (21K USDC for the second Milestone, 53K for the whole Grant). See Files changed

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [ ] **Level 2**:  Up to $30,000, 3 approvals
- [x] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (bank details via email _or_ BTC, Ethereum (USDC/DAI) or Polkadot/Kusama (USDT) address in the application).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Comment by Lord-Nymphis** [comment]
Let me know your thoughts.

File: applications/Datagen_Project.md [hunk 1]
```diff
@@ -245,3 +245,93 @@ The goal is to achive a fully functional mechanism for the random selection of t
...
+efficient communications between Solochains and Paracahains (which are the backbone of the Polkadot/Kusama ecosystem), is greatly
+reducing the use of Solochains as a complement for Parachains.
+
+h) Besides the importance of the development of the Milestone itself, we remember that Datagen (https://github.com/Datagen-Project/Whitepaper-2.0 ), as a service blockchain for verification of random computational problems, can greatly benefit the Polkadot
+network, by adding a tool to randomly verify on-chain off-chain computational workloads and to assign workloads in a decentralized way
+(as a long term solution on the vexing problem of having Polkadot’s and Kusama’s validators all hosted by AWS and similar providers
+with censorship capability over the network’s validators); as so, beyond the scope of the Milestone2, we think that progressing as fast
+as possible with the general roadmap of the Datagen network can be beneficial for the whole ecosystem.
+
+For all the reasons above, I would like to ask an integration of the amount due for the M2 deliverable of 9K USDC (from 12K to 21K
```
Comment (keeganquigley) [inline applications/Datagen_Project.md:334]: Thanks @Lord-Nymphis I'm not sure what is meant here by "ask an integration of the amount due..." are you asking for us to pay for the milestone before it is completed? Or can you explain what you mean here?

Please note in our [terms and conditions](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/T%26Cs.md), that we aren't able to pay for milestones before the delivery is evaluated and accepted.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the PR. Could you also update the [total costs](https://github.com/w3f/Grants-Program/blob/master/applications/Datagen_Project.md#overview-1) as well as the costs of the [milestone](https://github.com/w3f/Grants-Program/blob/master/applications/Datagen_Project.md#milestone-2--connecting-the-two-blockchains) as part of your PR?

File: applications/Datagen_Project.md (hunk 1 above)
Comment (Lord-Nymphis) [reply applications/Datagen_Project.md:334]: @keeganquigley No, OfC I don't mean for the Milestone to be paid before is completed

Comment (keeganquigley) [reply applications/Datagen_Project.md:334]: Thanks @Lord-Nymphis can you explain what you mean here?

Comment (Lord-Nymphis) [reply applications/Datagen_Project.md:334]: @keeganquigley I mean that, given that the end goal of the Datagen project (not the Milestone, which concerns only the RPC communication, between a Parachain and a Solochain, and not the Grant, which is just a PoC) is to provide cloud computing also for validators; the sooner we develop, the sooner we can have a positive eternality for other projects, including projects in the Polkadot ecosystem.
Obviously it doesn't mean that once we have developed this milestone the Datagen network is operational, but it means that we are one step faster.

Meaning that, by developing faster, we can achieve the goal of having a positive effect on other projects.

**Comment by Lord-Nymphis** [comment]
Ciao @Noc2 , I did their the you requested directly at the link you sent me (and in the abstract above). 
Let me know if it's ok.

**Comment by Noc2** [comment]
> Ciao @Noc2 , I did their the you requested directly at the link you sent me (and in the abstract above). Let me know if it's ok.

Could you please update it as part of this PR?

**Comment by Lord-Nymphis** [comment]
ok, I think that now it is updated

**Comment by keeganquigley** [comment]
Thanks for the changes @Lord-Nymphis I will mark it as ready for review and ping the rest of the committee.

**Comment by Lord-Nymphis** [comment]
thanks @keeganquigley

File: applications/Datagen_Project.md (hunk 1 above)
Comment (semuelle) [reply applications/Datagen_Project.md:334]: @keeganquigley, I think `integration` should be `increase`. Then it makes sense.

Comment (Lord-Nymphis) [reply applications/Datagen_Project.md]: @semuelle Thanks, done. And corrected few other typos as well.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the PR. But after looking at it again. I'm personally not willing to accept the amendment. The main reason is that I think the grant is against our guidelines, specifically: 
- "We do not award grants for projects that have been the object of a successful token sale." 
- and "Applications must not mention a specific token". 

This was already highly controversial when you [applied](https://github.com/w3f/Grants-Program/pull/980#issuecomment-1180755311), but now you even publicly said that you paid your devs working on the project in the `native token allocation.`

**Comment by Lord-Nymphis** [comment]
Hi @Noc2 

As said we tried to ask for the amendment because the guys from Parity told us it was worth a try. If you want to vote no or to close the pr is fine.

Just a clarification. I think you are referring to: _"....We thought we could count on @viac92 as an affordable dev. @viac92 was working at a much lower salary than the average substrate dev, both because he was involved in the project with a native token allocation and because he was very junior, and we paid him a salary during his formation in the substrate framework.  ...."_

I was not mentioning native token allocation for the milestone development. I already explained this fact, when we did the application, that devs already had done some previous work for the project (he developed the tokenomic smart contracts BEFORE the grant). For this reason, he had a pre-existing allocation in the project, as an early team member, and compounding the fact that he was an early team member with tokens (fact uncorrelated to the grant, as said in the original grant the tokens are used for other purposes that are not correlated to the object of the grant) and the fact that we paid him for months a salary in order to learn substrate (this being instead correlated to the grant) contributed to him accepting a lower salary than the benchmark. I never ever said that we paid him in native FOR the milestone or for the grant.

Businesses are complicated to run, devs do more than one thing for a company, there are many things going over, and it's not all black or white, my comment was just to give more context about the milestone development. BTW we both know which is a substrate dev salary, which is why I preferred to give more context.

As said, the scope of this amendment was to align the milestone with the real costs of the development.
I don't think in any way that the way in which we managed the dev payment is against the grant rules (we are an incorporated company, not an informal team just working on the grant and the relationship between devs and companies is complex), having said that, I'm fully aware that the process to accept amendments it's decentralized and it is fully within the rights of the commission members to accept or refuse our proposal for changes to what is already approved. We are totally on the same page on this, I just want to clarify eventual factual errors or misinterpretations.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the detailed answer. You are correct that the rule is somehow subjective and hard to define. So the rest of the team/committee might have a completely different opinion here and I will ping them again.

**Comment by keeganquigley** [comment]
Hi @Lord-Nymphis it is my opinion that if additional development work is needed, it should be added into the application. Without an expansion in scope I find it hard to warrant an increase. Therefore I won't approve the amendment as it stands. Let's see if the rest of the committee has a different opinion.

**Pull request closed without merging**